### ISSUE TITLE
Adding a GIF makes the whole machine freeze

### ISSUE THREAD

**Issue by niveK77pur** [jliljebl/flowblade#800]
Please include the following information.
1) Flowblade version (Help->About): 2.2.0
2) MLT version (Help->Runtime Environment): 6.0.0
3) Your distribution (Ubuntu, Debian, Mint etc.): Ubuntu 16.04.06 LTS
4) If possible, try to provide step-by-step instructions on how to reproduce the issue. From developer point of view this is the easieast way to understand what is happening.:

When I tried to add a GIF image using `add Video, Audio or Image ...` I noticed that flowblade weirdly stutters in each of the following cases. I had to force shut down my laptop **3 times now** to make sure it is because of the GIF. Here is what happened:

1) I tried adding a GIF, put it in the timeline and when I wanted to play back what I have, flowblade suddenly froze. My mouse wasn't responding anymore and when I wanted to check the terminal to see if there is any funny output it didn't work. The whole desktop just froze. I tried doing `ctrl+alt+F1` to try to kill flowblade from there but not even that was possible. Just **everything** was frozen. So I force shut down my laptop.
2) after rebooting I again tried adding the gif, but tried previewing it first by double clicking on it. And then the same happened. My computer was completely forzen. Again force shutdown.
3) Lastly I just added the gif and went back to the terminal immediately to see if there is anything suspicious ... however everything seemed fine (the last thing it said was `Saving project...`) so I tried to go back to flowblade and there it was; completely frozen again. Another force shutdown.

From what I deduced with this is that flowblade makes the whole machine freeze merely by adding a GIF. I would like to include some error messages from the terminal here but as I said, my computer was completely frozen and not responding to anything.

**Comment by smolnp** [comment]
My system does not freeze. The monitor plays from the Media tab. After adding a clip to Timeline, Playhead stops working.
![безымянный](https://user-images.githubusercontent.com/26217756/72211295-dd165f80-34d9-11ea-942f-2b90ada04132.png)
Added a color clip to Timeline, everything works, but the Monitor from Timeline when viewing GIF shows a white screen.
![ДОБАВИТЬ ПОСЛ](https://user-images.githubusercontent.com/26217756/72211301-ed2e3f00-34d9-11ea-9b86-070af69f4ab9.png)
вывод терминала:
Saving project...
[gif @ 0x7f74b0020400] Value 1.000000 for parameter 'seekable' out of range [-1 - 0]
[gif @ 0x7f74b0007840] Value 1.000000 for parameter 'seekable' out of range [-1 - 0]
[image2 @ 0x7f7490000900] Using AVStream.codec to pass codec parameters to muxers is deprecated, use AVStream.codecpar instead.
Saving project...

**Comment by niveK77pur** [comment]
Huh that is interesting ... I retried adding the same GIF as I was working with to a new blank project and it froze again. However, I also tried adding a gif that was shorter and it didn't quite freeze to death.

When I imported the GIF it started lagging really badly and occasionally froze for up to 5 seconds. It took what felt like a minute until my computer was responsive again but it wasn't quite working still. Only once I exited flowblade did everything get back to normal.

And funny enough, the terminal output didn't display anything either (but there is also no gif related warning like in your case).
```
MLT detection succeeded, 152 formats, 84 video codecs and 68 audio codecs found.
415 MLT services found.
Loading render profiles...
Loading filters...
MLT service dynamic_loudness not found.
MLT service ladspa.1210 not found.
Brightness dropped, MLT version too high for this filter.
Brightness Keyframed dropped, MLT version too high for this filter.
MLT service avfilter.colorchannelmixer not found.
MLT service avfilter.lut3d not found.
MLT service avfilter.lenscorrection not found.
MLT service avfilter.perspective not found.
Loading transitions...
RGB Adjustment dropped for Color Adjustment
Hue dropped for Color Adjustment
Gamma dropped for Lift Gain Gamma
G'MIC found
Player initialized with profile:  DV/DVD PAL
Selected color detected
BG color detected
Create SDL1 consumer...
Autosave started...
Saving project...
No user folders actions needed.
[swscaler @ 0x7fa9fc23f2e0] Warning: data is not aligned! This can lead to a speedloss
Saving project...

Exiting app...
```
(I am using i3wm btw if that makes any difference)

I don't know if it is because you use flowblade 2.4 or maybe even your system, but for me there is definitely an issue when adding GIFs ... now I am clueless as to what may cause this

**Comment by jliljebl** [comment]
I tested animated GIFs Flowblade, KDEnlive ans Shotcut. None worked correctly, KDEnlive and Shotcut showed some movement but not what was intended and what Firefox showed.

We will need to use PIL, ffmpeg or Imagemagik to do the convert to video on load.

Not really interested in doing GIF output, Imagemagic at least can create GIFs from PNG sequences.

I'm also not looking forward to adding new dependencies for this feature because an user able to do some rudimentary googling is able to get everything they need done using available command line programs.

I'll do some test on *ffmpeg* and  *PIL* to see if we can import with existing dependencies.

WORKAROUND: Use Imgamagik *convert* program or *ffmpeg* to create video from animated GIFs, and to create GIFs from rendered PNG sequences.

**Comment by jliljebl** [comment]
MLT does not support animated GIFs so I disallowed loading animated GIFs. Closing as wontfix as just refusing to load problematic media can't really be called as fix.

If MLT works animated with GIFs in the future I will allow adding them as media items.
